src/main/scala/com/github/takezoe/solr/scala/QueryBuilderBase.scala

```diff
@@ -266,7 +266,35 @@ trait QueryBuilderBase[Repr <: QueryBuilderBase[Repr]] {
           field.getValues.asScala.map { value => (value.getName, value.getCount) }.toMap
       )}.toMap
     }
-    MapQueryResult(numFound,numGroupsFound, queryResult, groupResult, facetResult,response.getQTime)
+    val facetPivotResult = response.getFacetPivot match {
+      case null => Map.empty[String, List[FacetPivot]]
+      case facetPivots =>
+        facetPivots.asScala.map { field =>
+          (
+            field.getKey,
+            field.getValue.asScala.map { field =>
+              FacetPivot(field.getField, field.getCount, toFacetPivotList(field.getPivot), field.getValue)
+            }.toList
+          )
+        }.toMap
+    }
+    MapQueryResult(numFound,numGroupsFound, queryResult, groupResult, facetResult,facetPivotResult,response.getQTime)
+  }
+
+  /**
+    * Convert java list of Pivot fields to scala list of facetPivots
+    * @param pivotFields
+  **/
+  def toFacetPivotList(pivotFields: java.util.List[PivotField]): List[FacetPivot] = {
+    if(pivotFields == null)
+      return null;
```

Comment:
`List.empty[FacetPivot]` might be better. What do you think?

Reply:
I think both of them would work but it seems like List.empty[FacetPivot] may save NPE.

Reply:
Since the documentation of Solrj is not clear about getFacetPivot method. Hence I am adding one extra check to see it returns either null or empty list